pkg/controller/factory/factory.go

```diff
@@ -279,12 +279,6 @@ func (f *Factory) ToController(name string, eventRecorder events.Recorder) Contr
 		cacheSyncTimeout:   defaultCacheSyncTimeout,
 	}
 
-	// Warn about too fast resyncs as they might drain the operators QPS.
-	// This event is cheap as it is only emitted on operator startup.
-	if c.resyncEvery.Seconds() < 60 {
-		ctx.Recorder().Warningf("FastControllerResync", "Controller %q resync interval is set to %s which might lead to client request throttling", name, c.resyncEvery)
-	}
-
```

Comment:
Yes, I was also considering amending line `284`. However I had a couple of issues with this approach. 

1. Creating a controller doesn't guarantee it will be run. So we might emit events for controller that will never be run.
2. The logic that handles `resyncEvery` interval is split between the factory and the base controller. I would prefer to have this logic in one place.

Does it make sense?

Reply:
I'm OK with that. Is the new boolean field necessary given typical usage patterns?

Reply:
I think it is necessary if we want to avoid repeating `FastControllerResync` event which happen on a controller restart.
I'm not sure if restarts are common operation. I suspect it is rather occasional.

I'm okay with removing the new boolean and report or every call to the `Run` method.

Reply:
Thanks. My secret motivation is to avoid worrying about data races for users that try to call Run concurrently.